hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -225,6 +225,22 @@ protected Dialect() {
 		defaultSizeStrategy = new DefaultSizeStrategyImpl();
 	}
 
+	private void logDeprecation() {
+		LOG.usingDialect( this );
+
+		Class<? extends Dialect> dialect = getClass();
+		if ( dialect.isAnnotationPresent(Deprecated.class) ) {
```

Comment:
This is against the Quarkus mandate that thou shall not read annotations at runtime.  Just saying

Reply:
Yeah I wondered about that... but this warning isn't necessary in Quarkus, so perhaps it doesn't matter?

Reply:
The warning is not, but the isAnnotationPresent would still happen.  I don't necessarily care, just pointing it out

Reply:
@Sanne what do you think here?

Reply:
Currently the constructor of Dialect is somethign which is not invoked at runtime, so this shouldn't be a problem.

I do wonder though if this method isn't better off to be invoked explicitly from the Dialect factory, rather than being a side effect of the constructor - I believe that would give us some more flexibility regarding wanting such a check to happen.

Reply:
Personally I like the "do this in `DialectFactory`" approach better, but if its not a problem for Quarkus then its not a big deal to me.

Reply:
I'm perfectly happy to do it in `DialectFactory`, my only minor concern would be is that if some people have code where the instantiate their `Dialect` directly, outside of the `DialectFactory` infrastructure, then they might not get the warning. On the other hand, if they're doing that, then I guess they would get a compilation warning in their code. So that's probably good enough.

Reply:
Let me know what you guys think and I'll make the change if that's what you prefer.